backend/users/views/user.py

```diff
@@ -0,0 +1,81 @@
+from django.conf import settings
+from django.contrib.auth.models import User
+
+from rest_framework import status
+from rest_framework.permissions import AllowAny
+from rest_framework.viewsets import GenericViewSet
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework_simplejwt.authentication import AUTH_HEADER_TYPES
+from rest_framework_simplejwt.exceptions import TokenError, InvalidToken, AuthenticationFailed
+
+from users.serializers import CustomTokenObtainPairSerializer
+
+import tldextract
+
+
+class UserViewSet(GenericViewSet):
```

Comment:
Using GenericViewSet here seems to add a log of complexity not only in this views file, but in the whole module. 

We could instead use a basic APIView. This way would wouldn't embark the whole CRUD system below the GenericViewSet.

Reply:
`GenericAPIView` is needed because `serializer_class` and `queryset` is used here.

```
class GenericViewSet(ViewSetMixin, generics.GenericAPIView):
    """
    The GenericViewSet class does not provide any actions by default,
    but does include the base set of generic view behavior, such as
    the `get_object` and `get_queryset` methods.
    """
    pass
```

`get_serializer` is part of `GenericAPIView`
```
    def get_serializer(self, *args, **kwargs):
        """
        Return the serializer instance that should be used for validating and
        deserializing input, and for serializing output.
        """
        serializer_class = self.get_serializer_class()
        kwargs['context'] = self.get_serializer_context()
        return serializer_class(*args, **kwargs)
```

Reply:
I still think we could have gone without the overhead, but I see your point. Let's keep it this way for now.

I guess this was my last comment about the code, I'll move on to testing and then PR approvals. We're almost there!